src/Data/List/Membership/Propositional/Properties.agda

```diff
@@ -362,28 +361,27 @@ finite inj (x ∷ xs) fᵢ∈x∷xs = ¬¬-excluded-middle helper
   helper (yes (i , fᵢ≡x)) = finite f′-inj xs f′ⱼ∈xs
     where
     f′ : ℕ → _
-    f′ j with does (i ≤? j)
-    ... | true  = f (suc j)
-    ... | false = f j
+    f′ j with i ≤? j
+    ... | yes _ = f (suc j)
+    ... | no  _ = f j
```

Comment:
The last time I looked at the `Dec` set-up this change would mean that we now compute the proof, even when we don't need it. For example with naturals, the boolean decision procedure complexity is `O(max(log(i),log(j)))` because we use built-in arithmetic whereas the proof i s`O(max(i), max(j))` because we use unary arithmetic. 

Is this still the case?

Reply:
Good question!
I was focused on refactoring the 'heavy' `with`/case-analysis in the proof of `f′-injective′`, and not thinking about the computational content of `f′` itself...
... but actually that's OK, because the whole `where` block is dedicated to proving a *negative* statement, so nothing in it has/will have computational content! (And even if it did, the final `¬¬-excluded-middle` wrapper clinches it)

Reply:
Okay, sounds good!